swift/ql/lib/codeql/swift/generated/ParentChild.qll

```diff
@@ -0,0 +1,2182 @@
+        result = e.(RepeatWhileStmt).getImmediateBody() and
+        partialAccessor = "Body()"
+      )
+    )
+    or
+    exists(int n, int nBase |
+      n = 0 + getMaximumChildrenIndexApplyExpr(e) and
+      nBase = n + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(SelfApplyExpr).getImmediateBase() and partialAccessor = "Base()"
+      )
+    )
+    or
+    exists(int n, int nArgument |
+      n = 0 + getMaximumChildrenIndexLookupExpr(e) and
+      nArgument =
+        n + 1 + max(int i | i = 0 or exists(e.(SubscriptExpr).getImmediateArgument(i)) | i) and
+      (
+        none()
+        or
+        result = e.(SubscriptExpr).getImmediateArgument(index - n) and
+        partialAccessor = "Argument(" + (index - n).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nExpr, int nCase |
+      n = 0 + getMaximumChildrenIndexLabeledStmt(e) and
+      nExpr = n + 1 and
+      nCase = nExpr + 1 + max(int i | i = 0 or exists(e.(SwitchStmt).getImmediateCase(i)) | i) and
+      (
+        none()
+        or
+        index = n and result = e.(SwitchStmt).getImmediateExpr() and partialAccessor = "Expr()"
+        or
+        result = e.(SwitchStmt).getImmediateCase(index - nExpr) and
+        partialAccessor = "Case(" + (index - nExpr).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nBody |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nBody = n + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(GuardStmt).getImmediateBody() and partialAccessor = "Body()"
+      )
+    )
+    or
+    exists(int n, int nThen, int nElse |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nThen = n + 1 and
+      nElse = nThen + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(IfStmt).getImmediateThen() and partialAccessor = "Then()"
+        or
+        index = nThen and result = e.(IfStmt).getImmediateElse() and partialAccessor = "Else()"
+      )
+    )
+    or
+    exists(int n, int nParam |
+      n =
+        0 + getMaximumChildrenIndexAbstractStorageDecl(e) + getMaximumChildrenIndexGenericContext(e) and
+      nParam = n + 1 + max(int i | i = 0 or exists(e.(SubscriptDecl).getImmediateParam(i)) | i) and
+      (
+        none()
+        or
+        result = e.(SubscriptDecl).getImmediateParam(index - n) and
+        partialAccessor = "Param(" + (index - n).toString() + ")"
+      )
+    )
+    or
+    exists(int n, int nBody |
+      n = 0 + getMaximumChildrenIndexLabeledConditionalStmt(e) and
+      nBody = n + 1 and
+      (
+        none()
+        or
+        index = n and result = e.(WhileStmt).getImmediateBody() and partialAccessor = "Body()"
+      )
+    )
+  }
+}
+
+/**
+ * Gets the "immediate" parent of `e`. "Immediate" means not taking into account node resolution: for example
+ * if `e` has conversions, `getImmediateParent(e)` will give the bottom conversion in the hidden AST.
```

Comment:
Should I read "bottom conversion" as the outer-most conversion?

Reply:
to prove the point, it's actually the innermost conversion. In C++ terms, for
```
(double)(int)x.i
```
* `getImmediateParent` called on the `.i` accessor would give uncoverted `x.i`
* `getImmediateParent` called on `x.i` would give the innermost `(int)x.i`